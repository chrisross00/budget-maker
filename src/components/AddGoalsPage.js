import React from 'react';
import GoalsForm from './GoalsForm';
import { connect } from 'react-redux';
import { addGoal } from '../actions/goal';

export class AddGoalsPage extends React.Component {
  onSaveGoal = (goal) => {
    this.props.addGoal(goal);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add your goals</h1>
          </div>
        </div>
        <GoalsForm
          onSaveGoal={this.onSaveGoal} />
      </div>
    )
  }
}


const mapDispatchToProps = (dispatch) => ({
  addGoal: (goal) => { dispatch(addGoal(goal)) }
})

export default connect(undefined, mapDispatchToProps)(AddGoalsPage)
