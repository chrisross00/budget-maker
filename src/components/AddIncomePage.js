import React from 'react';
import { connect } from 'react-redux';
import { addIncome, updateIncome } from '../actions/income';
import IncomeForm from './IncomeForm';
import List from './List';

import { FormHeader } from './FormHeader';
import formatInUsd from '../helpers/formatInUsd';


export class AddIncomePage extends React.Component {
  addIncome = (income) => {
    this.props.addIncome(income);
    this.props.history.push('/');
  }
  updateIncome = (id, income) => {
    this.props.updateIncome(id, income);
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Enter your income</h1>
          </div>
        </div>
        <div className="content-container shadow">
          <FormHeader
            formType={'income'} />
          <IncomeForm
            addIncome={this.addIncome}
            updateIncome={this.updateIncome}
            income={this.props.income} />
          <List
            isOpened={true}
            parent={'Income'}
            propsToRender={this.props.income}
            wordToRender={'Income'} />
        </div>
      </div>
    );
  };
};

const mapStateToProps = (state) => {
  return {
    income: state.income
  }
}
const mapDispatchToProps = (dispatch) => ({
  addIncome: (income) => dispatch(addIncome(income)),
  updateIncome: (id, income) => dispatch(updateIncome(id, income))
});


export default connect(mapStateToProps, mapDispatchToProps)(AddIncomePage);
