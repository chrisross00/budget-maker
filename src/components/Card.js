import React from 'react';

export const Card = (WrappedFormComponent) => {
  return (props) => (
    <div className="content-container shadow">
      <WrappedFormComponent {...props} />
    </div>
  )
}

export default Card;
