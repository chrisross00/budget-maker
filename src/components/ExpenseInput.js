import React from 'react';
import { connect } from 'react-redux';
import { addExpense, updateExpense } from '../actions/expense';

class ExpenseInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expenseType: this.props.expense ? this.props.expense.expenseType : '',
      expenseCategory: this.props.expense ? this.props.expense.expenseCategory : '',
      expenseCategoryId: this.props.expense ? this.props.expense.expenseCategoryId : 0,
      normalAmount: this.props.expense ? (this.props.expense.normalAmount / 100).toString() : '',
      changed: false,
      committed: false
    }
  }
  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ normalAmount: amount, changed: true }));
    }
  }
  onSubmit = (e) => {
    e.preventDefault();
    let holder = [];
    let newExpense = {
      expenseType: this.props.expenseTypeId,
      expenseCategory: this.props.expenseCategory,
      expenseCategoryId: this.props.expenseCategoryId,
      normalAmount: this.state.normalAmount,
    }
    if (this.props.expenses.length !== 0) {
      this.props.expenses.map((expense) => {
        holder.push(expense.expenseCategoryId);
      });
      if (holder.includes(this.props.expenseCategoryId)) {
        this.props.updateExpense(this.props.expenseCategoryId, newExpense);
      } else {
        this.props.addExpense(newExpense);
      }
    } else {
      this.props.addExpense(newExpense);
    }
    this.setState(() => ({
      changed: false,
      committed: true
    }))
  };
  render() {
    const updateSaveWord = (this.state.committed) ? "Update" : "Save"
    return (
      <div className="form">
        <input
          className="text-input"
          type="number"
          placeholder={`How much do you pay for ${this.props.label}?`}
          id={this.props.expenseCategoryId}
          onChange={this.onAmountChange}
          value={this.state.normalAmount} />
        {this.state.normalAmount
          ? (<button className="button"
            onClick={this.onSubmit}
            disabled={!this.state.changed}>{updateSaveWord} Expense</button>)
          : ''}
      </div>
    )
  };
}

const mapStateToProps = (state) => {
  return {
    expenses: state.expense
  }
};

const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => { dispatch(addExpense(expense)) },
  updateExpense: (id, expense) => { dispatch(updateExpense(id, expense)) }
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseInput);
