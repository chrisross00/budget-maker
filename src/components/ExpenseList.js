import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';

const ExpenseList = ({ expense }) => (
  <div>
    <div className="list-header">
      <div className="show-for-desktop">Monthly Expenses</div>
      <div className="show-for-mobile">Monthly Expenses</div>
      <div className="show-for-desktop">Amount</div>
    </div>
    <div className="list-body">
      {expense.length === 0
        ? (
          <div className="list-item list-item-message">
            <span>No expenses added yet</span>
          </div>
        )
        : (expense.map((expense) => {
          return <ExpenseListItem {...expense} key={expense.id} />
        })
        )
      }
    </div>
  </div>
);

const mapStateToProps = (state) => ({
  expense: state.expense
});

export default connect(mapStateToProps)(ExpenseList)
