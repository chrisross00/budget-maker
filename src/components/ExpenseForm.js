import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ExpenseInput from './ExpenseInput';

export class ExpenseForm extends React.Component {
  render() {
    return (
      <form
        autoComplete="off"
        className="form">
        {this.props.error && <p className="form__error">{this.props.error}</p>}
        {
          this.props.expenseCategory.map(
            (item) => {
              return <ExpenseInput
                key={item.expenseCategoryId} {...item} />
            })
        }
        <div>
          <Link
            className="button"
            to="/">
            Save All Expenses</Link>
        </div>
      </form>
    );
  };
};

const mapStateToProps = (state) => {
  return {
    expenseCategory: state.expenseCategory
  }
};

export default connect(mapStateToProps)(ExpenseForm);
