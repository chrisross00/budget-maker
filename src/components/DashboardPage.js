import React from 'react';
import { Link } from 'react-router-dom';
import IncomeList from './IncomeList';
import ExpenseList from './ExpenseList';
import GoalList from './GoalList';
import SummaryList from './SummaryList';

export const ExpenseDashboardPage = () => (
  <div className="content-container">
    <SummaryList />
    <Link
      to="/income">
      <h2>Step 1. Add your income</h2>
      <hr />
    </Link>
    <IncomeList />
    <Link
      to="/expenses">
      <h2>Step 2. Add your monthly expenses</h2>
    </Link>
    <ExpenseList />
    <Link
      to="/goals">
      <h2>Step 3. Define your goals</h2>
    </Link>
    <GoalList />
  </div>
);

export default ExpenseDashboardPage;
