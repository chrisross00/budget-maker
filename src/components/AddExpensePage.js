import React from 'react';
import ExpenseForm from './ExpenseForm';

export class AddExpensePage extends React.Component {
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">Enter your expenses</h2>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  };
};


export default AddExpensePage;
