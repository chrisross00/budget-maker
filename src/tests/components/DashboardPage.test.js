import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseDashboardPage } from '../../components/DashboardPage';

test('should render DashboardPage component properly', () => {
  const wrapper = shallow(<ExpenseDashboardPage />);
  expect(wrapper).toMatchSnapshot();
});
