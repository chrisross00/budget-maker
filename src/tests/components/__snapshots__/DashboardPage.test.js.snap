// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage component properly 1`] = `
<div
  className="content-container"
>
  <Link
    replace={false}
    to="/income"
  >
    <h2>
      Step 1. Add your income
    </h2>
  </Link>
  <Connect(IncomeList) />
  <Link
    replace={false}
    to="/expenses"
  >
    <h2>
      Step 2. Add your monthly expenses
    </h2>
  </Link>
  <Connect(ExpenseList) />
  <Link
    replace={false}
    to="/goals"
  >
    <h2>
      Step 3. Define your goals
    </h2>
  </Link>
  <Connect(GoalList) />
  <hr />
  <Connect(Summary) />
</div>
`;
